ts/features/payments/checkout/utils/index.ts

```diff
@@ -78,3 +79,22 @@ export const isDueDateValid = (date: string): string | undefined => {
   tenYearsFromNow.setFullYear(tenYearsFromNow.getFullYear() + YEARS_TO_EXPIRE);
   return new Date(date) > tenYearsFromNow ? undefined : formattedDate;
 };
+
+export const getSubCategoryFromFaultCode = (data: any, statusCode: string) => {
+  // check if there is a subcategory array that includes passed element
+  const subcategoryKey = Object.keys(data.payments.subcategories).find(key =>
+    data.payments.subcategories[key].includes(statusCode)
+  );
+  // if there is, return the mapped subcategory with the zendesk category id
+  if (subcategoryKey) {
+    return {
+      value: subcategoryKey,
+      zendeskCategoryId: data.payments.subcategoryId
+    };
+  }
+  // if not, return the parent category
+  return {
+    value: "pagamenti_pagopa",
+    zendeskCategoryId
+  };
```

Comment:
As mentioned above, if both the `categoryId` and the` subCategoryId` need to be mapped, I would suggest returning a nullish value here if no subcategory is found. Then, during the association of the subcategory to the ticket, you could check if the subcategory is null. If it’s not null, add it to the ticket; otherwise, omit it